src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+
+In essence, we are now going to try and think about these crossings as if they are independent two state systems whose states and couplings we have to determine.
+
+Let's see how we get on.
+
+
+
+### 4.2.2 - Crossings and anti-crossings
+
+We're now going to switch on the interaction term in the Hamiltonian. In order to more clearly visualise things, we will choose a stronger coupling than we did in the previous tutorial - specifically $U=0.2$.
+
+```python
+df = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[-0.5,5.5],legend=True, 
+        title=f"Stationary states for {H_latex}     ($\omega=1$, $U=0.2$)     (Fig 2)");
+plt.ylabel("Energy");
+```
+
+There are many things to say about what we see in Fig 2.
+
+Two main features are:
+1. the splitting of the levels increases with increasing level number i.e increasing number of bosons. 
+2. not all crossings have been split apart into anti-crossings as we expected - this means some levels don't interact with each other at all. 
+
+On 1. Applying our knowledge from Tutorial 1, we would say that the effective coupling between levels (which is proportional to the level splitting) increases with increasing boson number.
+
+On 2. Upon closer inspection, we can see that the level splittings only occur when $\Delta E  \approx n \omega$ where n is an **odd** integer (we'll come to why we now use $\approx$ instead of = shortly). This tells us that our prediction of other resonances was only correct for the odd integer ones - we will understand why soon. To see the anti-crossing more clearly for $\Delta E \approx 3\omega$ we need to zoom in a bit.
+
+To perform this zoom, it is best to perform a higher resolution scan of $\Delta E$:
+
+```python
+df = make_df_for_energy_scan("$\Delta E$", 2.8, 2.95, 201, two_state.shape[0])
+
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[1.9,2],legend=True, 
+        title=f"Stationary states for {H_latex}    ($\omega=1$, $U=0.2$)     (Fig 3)");
+plt.ylabel("Energy");
+
+
+```
+
+The level splitting seen in Fig 3 is much smaller than those seen in Fig 2 when $\Delta E \approx \omega$ - i.e. the effective coupling between levels at this resonance is much less. This gives some physical meaning to our earlier description of these other resonances as "non-primary".
```

Comment:
Wah, I'm confused. I thought the smaller the gap, the stronger the coupling....

Reply:
I also was confused but remember the last plot by byrnes
https://coldfusionblog.net/2017/07/09/numerical-spin-boson-model-part-1/

![image](https://user-images.githubusercontent.com/28784964/82221453-4db60880-9920-11ea-9db6-8a9fbb851ebc.png)

Reply:
Hmm, good point. I remember we discussed this before but maybe I never fully understood this aspect. Maybe let's briefly recap this tomorrow.

Reply:
I know what you mean - I think I might have caused this confusion when we talked about it a few weeks ago.

My thought was that the closer the levels are in energy (also frequency) then the stronger the interaction between them could be. This thought came out of the idea of classical pendulums - the closer the lengths are to each other then the closer the frequencies and so the stronger the interaction will be. 

That is all true. But, then we have to ask ourselves - ok these two pendulums have very similar frequencies so in principle if they were coupled by some string then we would expect something interesting to happen. But what is this interesting thing? Well, we find out that the interesting thing is that the original frequencies get pushed apart in a way that's proportional to the coupling between them.

Maybe said in a shorter way we could say there are two things at play here
1. The potential for interaction - inversely proportional to the energy difference (in the absence of coupling)
2. The result of coupling - frequency splitting of levels with high potential for interaction. The splitting is proportional to the coupling  strength